common/environment/environment.cpp

```diff
@@ -2615,6 +2614,41 @@ void getRoxieProcessServers(const char *process, SocketEndpointArray &servers)
     getRoxieProcessServers(queryRoxieProcessTree(root, process), servers);
 }
 
+extern ENVIRONMENT_API RoxieTargetType readRoxieTargetType(const char *roxieName)
+{
+    RoxieTargetType roxieTargetType = RTTUnknown;
+    if (isEmptyString(roxieName))
+        return roxieTargetType;
+
+    bool hasPortZero = false, hasNonZeroPort = false;
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactory(true);
+    Owned<IConstEnvironment> environment = factory->openEnvironment();
+    Owned<IPropertyTree> envRoot = &environment->getPTree();
+
+    VStringBuffer xpath("Software/RoxieCluster[@name='%s'][1]", roxieName);
+    Owned<IPropertyTreeIterator> farmers = envRoot->queryPropTree(xpath.str())->getElements("RoxieFarmProcess");
```

Comment:
defensive: This assumes roxieName has a matching entry in the environment.  If it doesn't for some reason the getElements() call will crash.

Reply:
Will add a check.